lib/utmp.c

```diff
@@ -318,16 +318,10 @@ prepare_utmp(const char *name, const char *line, const char *host,
 				struct sockaddr_in *sa =
 					(struct sockaddr_in *) info->ai_addr;
 # if defined(HAVE_STRUCT_UTMPX_UT_ADDR)
-				memcpy (&(utent->ut_addr),
-				        &(sa->sin_addr),
-				        MIN(sizeof(utent->ut_addr),
-				            sizeof(sa->sin_addr)));
+				utent->ut_addr = sa->sin_addr.s_addr;
```

Comment:
Looked over this for quite awhile.  The UTMPX_UT_ADDR{,_V6} #defines really make this uglier...  Are these required to support musl?

Anyway, I agree it's an improvement, thanks.

Reply:
I think I have some patch removing some of those.  I'll check.

Reply:
Thank you!  You've merged quite a lot of patches today.  :-)